<launch>

<node pkg="detection" type="detection" name="detection" output="screen">
    <param name="cameraFrame" value="camera_depth_optical_frame"/>
    <param name="cameraPoint" value="/camera/depth/points"/>
    <param name="odomFrame" value="/odom/ground_truth"/>
    <param name="baseFrame" value="/base"/>
    <param name="lidarFrame" value="/map"/>
    <param name="FilterFieldName" value="z"/>
    <param name="setFilterLimitsUp" value="5.0"/>
    <param name="setFilterLimitsDown" value="0.0"/>
    <param name="use_static_tf" value="false"/>
    <param name="elevation_wait_time" value="10"/>
</node>

<node pkg="detection" type="rawmapping" name="rawmapping" output="screen">
    <param name="cameraFrame" value="camera_depth_optical_frame"/>
    <param name="mapFrame" value="map"/>
    <param name="cameraPoint" value="/camera_point"/>
    <param name="FilterFieldName" value="z"/>
    <param name="setFilterLimitsUp" value="3.5"/>
    <param name="setFilterLimitsDown" value="0.0"/>
    <param name="setFilterheight" value="0.12"/>
    <param name="setOutlierFilter" value="false"/>
    <param name="setOutlierFilterMeanK" value="10"/>
    <param name="setOutlierFilterThresh" value="1.0"/>
    <param name="voxelsize" value="0.04"/>
    <param name="roshz" value="1000"/>
    <param name="wait_surfel_finished" value="10000"/>
    <param name="setPointNumLimit" value="250000"/>
</node>

<!-- Elevation mapping node -->
<node pkg="elevation_mapping" type="elevation_mapping" name="elevation_mapping" output="screen">
    <param name="length_in_x" value="20.0"/>
    <param name="length_in_y" value="20.0"/>
    <param name="resolution" value="0.1"/>
    <param name="threshold_roughness" value="0.2"/>
    <param name="threshold_height" value="0.2"/>
<!--        <param name="min_variance" value="0.000009"/>-->
    <param name="max_variance" value="0.04"/>
<!--        <param name="mahalanobis_distance_threshold" value="2.0"/>-->
<!--        <param name="min_horizontal_variance" value="0.09"/>-->
<!--        <param name="max_horizontal_variance" value="0.5"/>-->
    <param name="robot_pose_cache_size" value="200"/>
    <param name="min_update_rate" value="1.0"/>
    <param name="fused_map_publishing_rate" value="3.0"/>

    <param name="point_cloud_topic" value="/camera_point"/>
    <param name="robot_pose_with_covariance_topic" value="/odom_pose"/>
    <param name="base_frame_id" value="base"/>
    <param name="map_frame_id" value="map"/>
    <param name="track_point_frame_id" value="base"/>
    <param name="sensor_frame_id" value="camera_depth_optical_frame"/>
    <param name="robot_base_frame_id" value="base"/>

    <param name="enable_visibility_cleanup" value="false"/>
    <param name="visibility_cleanup_rate" value="0.05"/>
    <param name="scanning_duration" value="0.3"/>

    <param name="sensor_processor/type" value="structured_light"/>
    <param name="sensor_processor/cutoff_max_depth" value="3.5"/>
    <param name="sensor_processor/cutoff_min_depth" value="0.0"/>
    <param name="sensor_processor/normal_factor_a" value="0.000611"/>
    <param name="sensor_processor/normal_factor_b" value="0.003587"/>
    <param name="sensor_processor/normal_factor_c" value="0.3515"/>
    <param name="sensor_processor/normal_factor_d" value="0"/>
    <param name="sensor_processor/normal_factor_e" value="1"/>
    <param name="sensor_processor/lateral_factor" value="0.01576"/>

    <param name="sensor_processor/apply_voxelgrid_filter" value="true"/>
    <param name="sensor_processor/voxelgrid_filter_size" value="0.03"/>
</node>

<!-- Launch visualizations for the resulting elevation map -->
<include file="$(find elevation_mapping_demos)/launch/visualization.launch" />

</launch>
